scripts/presubmit.sh

```diff
@@ -17,16 +17,23 @@
 set -eEuo pipefail
 
 ROOT="$(cd "$(dirname "$0")/.." &>/dev/null; pwd -P)"
-SOURCE_DIRS="cmd internal tools"
-
+# Find all non-generated Go files. This works by excluding
+# files that start with the official "generated file" header.
+# See https://github.com/golang/go/issues/13560#issuecomment-288457920
+mapfile -t SOURCE_FILES < <( grep -L -R "^\/\/ Code generated .* DO NOT EDIT\.$$" --include="*.go" ${ROOT} )
```

Comment:
Is `mapfile` a common command? I don't seem to have it on my Mac.

Reply:
same.

Reply:
It's a [bash 4.0 (2009)](https://www.tldp.org/LDP/abs/html/bashver4.html) feature, I think it might be that the default version of bash shipping with Mac is ancient, but I understand it's practically unreasonably to expect all users to upgrade their bash version from the default one, so I'll see if I can find a workaround.

Reply:
Please try again, the new command should work better.